compiler/rustc_lint/messages.ftl

```diff
@@ -456,6 +456,10 @@ lint_invalid_nan_comparisons_eq_ne = incorrect NaN comparison, NaN cannot be dir
 
 lint_invalid_nan_comparisons_lt_le_gt_ge = incorrect NaN comparison, NaN is not orderable
 
+lint_invalid_null_arguments = calling this function with a null pointer is Undefined Behavior, even if the result of the function is unused
+    .note = null pointer originate from here
+    .suggestion = consider using a dangling pointer instead
```

Comment:
`ptr::null()` returns a dangling pointer, so this wording is ill-formed.

I'm not sure there is a useful suggestion to make, perhaps if you have surveyed sites this current warns and concluded that `ptr::dangling()` would be _correct_ (as opposed to just getting past the type checker), then it would be appropriate to say `ptr::dangling()`.

Reply:
I looked more closely at the GitHub search and I think you're right, recommending a dangling ptr (from `ptr::dangling()`, not `ptr::null()`) seems ill-advised in many cases.

I've removed the suggestion and redirected users to our documentation on [pointers](https://doc.rust-lang.org/std/ptr/index.html) instead.